### PR TITLE
Added Captain npc and fixed Player.isPremium missing check

### PR DESCRIPTION
**Pull request by emil92b**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

* Added Travel Npc "Captain" that takes you to (Trekolt, Rhyves, Varak and Saund). Also added chance for ghost pirate island
* fixed so players with "IsAlwaysPremium" flag can travel with npc etc

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/npc/scripts/ship.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+NpcSystem.parseParameters(npcHandler)
+
+function onCreatureAppear(cid)              npcHandler:onCreatureAppear(cid)            end
+function onCreatureDisappear(cid)           npcHandler:onCreatureDisappear(cid)         end
+function onCreatureSay(cid, type, msg)      npcHandler:onCreatureSay(cid, type, msg)    end
+function onThink()                          npcHandler:onThink()                        end
+
+keywordHandler:addKeyword({'captain'}, StdModule.say, {npcHandler = npcHandler, text = 'I am the captain of this sailing-ship.'})
+keywordHandler:addKeyword({'trip'}, StdModule.say, {npcHandler = npcHandler, text = 'Where do you want to go? To Trekolt, Rhyves, Varak or Saund?'})
+keywordHandler:addKeyword({'ice'}, StdModule.say, {npcHandler = npcHandler, text = 'I\'m sorry, but we don\'t serve the routes to the Ice Islands.'})
```
Comment (EPuncker) [inline data/npc/scripts/ship.lua:12]:
```suggestion
```

File: data/npc/scripts/ship.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    node1:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 100, destination = math.random(10) == 1 and {x=489,y=384,z=4} or {x=95,y=117,z=7}})
+    node1:addChildKeyword({'no'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, reset = true, text = 'We would like to serve you some time.'})
+
+local node2 = keywordHandler:addKeyword({'rhyves'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Rhyves for 120 gold?'})
+    node2:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 120, destination = {x=139,y=337,z=6}})
+    node2:addChildKeyword({'no'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, reset = true, text = 'We would like to serve you some time.'})
+
+local node3 = keywordHandler:addKeyword({'varak'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Varak for 150 gold?'})
+    node3:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 160, destination = {x=271,y=516,z=11}})
```
Comment (EPuncker) [inline data/npc/scripts/ship.lua:23]:
```suggestion
    node3:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 150, destination = Position(271, 516, 1)})
```

File: data/npc/scripts/ship.lua [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    node3:addChildKeyword({'no'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, reset = true, text = 'We would like to serve you some time.'})
+
+local node4 = keywordHandler:addKeyword({'saund'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Saund for 150 gold?'})
+    node4:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 150, destination = {x=258,y=602,z=7}})
```
Comment (EPuncker) [inline data/npc/scripts/ship.lua:27]:
```suggestion
    node4:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 150, destination = Position(258, 602, 7)})
```

File: data/npc/scripts/ship.lua [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+keywordHandler:addKeyword({'trip'}, StdModule.say, {npcHandler = npcHandler, text = 'Where do you want to go? To Trekolt, Rhyves, Varak or Saund?'})
+keywordHandler:addKeyword({'ice'}, StdModule.say, {npcHandler = npcHandler, text = 'I\'m sorry, but we don\'t serve the routes to the Ice Islands.'})
+
+local node1 = keywordHandler:addKeyword({'trekolt'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Trekolt for 100 gold?'})
+    node1:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 100, destination = math.random(10) == 1 and {x=489,y=384,z=4} or {x=95,y=117,z=7}})
+    node1:addChildKeyword({'no'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, reset = true, text = 'We would like to serve you some time.'})
+
+local node2 = keywordHandler:addKeyword({'rhyves'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Rhyves for 120 gold?'})
+    node2:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 120, destination = {x=139,y=337,z=6}})
```
Comment (EPuncker) [inline data/npc/scripts/ship.lua:19]:
```suggestion
    node2:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 120, destination = Position(139, 337, 6)})
```

File: data/npc/scripts/ship.lua [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+function onCreatureDisappear(cid)           npcHandler:onCreatureDisappear(cid)         end
+function onCreatureSay(cid, type, msg)      npcHandler:onCreatureSay(cid, type, msg)    end
+function onThink()                          npcHandler:onThink()                        end
+
+keywordHandler:addKeyword({'captain'}, StdModule.say, {npcHandler = npcHandler, text = 'I am the captain of this sailing-ship.'})
+keywordHandler:addKeyword({'trip'}, StdModule.say, {npcHandler = npcHandler, text = 'Where do you want to go? To Trekolt, Rhyves, Varak or Saund?'})
+keywordHandler:addKeyword({'ice'}, StdModule.say, {npcHandler = npcHandler, text = 'I\'m sorry, but we don\'t serve the routes to the Ice Islands.'})
+
+local node1 = keywordHandler:addKeyword({'trekolt'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Trekolt for 100 gold?'})
+    node1:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 100, destination = math.random(10) == 1 and {x=489,y=384,z=4} or {x=95,y=117,z=7}})
```
Comment (EPuncker) [inline data/npc/scripts/ship.lua:15]:
```suggestion
    node1:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, cost = 100, destination = math.random(10) == 1 and Position(489, 384, 4) or Position(95, 117, 7)})
```

**Review by EPuncker — CHANGES_REQUESTED** [review]

File: data/npc/scripts/ship.lua (hunk 1 above)
Comment (emil92b) [reply data/npc/scripts/ship.lua:12]: elaborate?

Comment (EPuncker) [reply data/npc/scripts/ship.lua:12]: I suppose this was a left over from other npc?

Comment (emil92b) [reply data/npc/scripts/ship.lua:12]: nah i added these so you can use as an example, but i did copy the texts from rl files, and i left out jarvik city as it looked unfinished, so thats why theres a 'ice' keyword in there

Comment (EPuncker) [reply data/npc/scripts/ship.lua:12]: I see

File: data/npc/scripts/ship.lua [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+function onCreatureAppear(cid)              npcHandler:onCreatureAppear(cid)            end
+function onCreatureDisappear(cid)           npcHandler:onCreatureDisappear(cid)         end
+function onCreatureSay(cid, type, msg)      npcHandler:onCreatureSay(cid, type, msg)    end
+function onThink()                          npcHandler:onThink()                        end
+
+keywordHandler:addKeyword({'captain'}, StdModule.say, {npcHandler = npcHandler, text = 'I am the captain of this sailing-ship.'})
+keywordHandler:addKeyword({'trip'}, StdModule.say, {npcHandler = npcHandler, text = 'Where do you want to go? To Trekolt, Rhyves, Varak or Saund?'})
+keywordHandler:addKeyword({'ice'}, StdModule.say, {npcHandler = npcHandler, text = 'I\'m sorry, but we don\'t serve the routes to the Ice Islands.'})
+
+local node1 = keywordHandler:addKeyword({'trekolt'}, StdModule.say, {npcHandler = npcHandler, onlyFocus = true, text = 'Do you seek a passage to Trekolt for 100 gold?'})
```
Comment (nekiro) [inline data/npc/scripts/ship.lua:14]: why are you initializing the value called node if it has no use?

Comment (emil92b) [reply data/npc/scripts/ship.lua:14]: care to explain what you mean by that? looks alright to me

Comment (nekiro) [reply data/npc/scripts/ship.lua:14]: local node1 = keywordHandler:addKeyword...

**Comment by DSpeichert** [comment]
@emil92b do you intent to continue working on this PR?

File: data/npc/scripts/ship.lua (hunk 6 above)
Comment (Znote) [reply data/npc/scripts/ship.lua:14]: @nekiro Each node has duplicate child keywords (`yes` and `no`) which does different actions depending on the node. (different position). 

But if keywordhandler returns itself/its node, perhaps these can be appended to the same line. 
```lua
keywordHandler:addKeyword({'rhyves'}, ... ):addChildKeyword({'yes'}, ... ):addChildKeyword({'no'}, ... )
-- beautified?
keywordHandler:addKeyword({'rhyves'}, ... )
    :addChildKeyword({'yes'}, ... )
    :addChildKeyword({'no'}, ... )
```

**Review by EPuncker — APPROVED** [review]

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**